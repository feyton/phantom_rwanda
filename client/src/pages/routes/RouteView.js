import React from 'react';
import { MapContainer, Marker, Popup, TileLayer } from 'react-leaflet';
import { useLocation } from 'react-router-dom';

const RouteView = () => {
  const location = useLocation();
  const routeInfo = location.state?.route;

  return (
    <>
      {routeInfo ? (
        <div className="p-4 bg-white w-[90vh] font-raleway">
          <h2 className="text-2xl text-center font-bold">Route informations</h2>
          <div className="mt-5">
            <ul>
              <li>
                <b> Origin: </b> {routeInfo.origin}
              </li>
              <li>
                <b> Destination: </b> {routeInfo.destination}
              </li>
            </ul>
          </div>
          <div className=" h-[300px] overflow-hidden py-2">
            <h1 className="font-raleway font-bold"> Map </h1> <hr />
            <MapContainer
              center={{ lat: -1.936671, lng: 30.053524 }}
              zoom={13}
              zoomControl={false}
              className="h-full w-full mb-2"
            >
              <TileLayer
                attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
                url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
              />
              <Marker position={[-1.936671, 30.053524]}>
                <Popup>
                  Route from <b>{routeInfo.origin}</b> to
                  <b>{routeInfo.destination}</b>.
                </Popup>
              </Marker>
            </MapContainer>
          </div>
        </div>
      ) : (
        <div className=" p-12">
          <h1>Route not found</h1>
        </div>
      )}
    </>
  );
};

export default RouteView;
