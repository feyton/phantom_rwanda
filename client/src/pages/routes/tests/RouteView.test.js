import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import RouteView from '../RouteView.js';

const mockedUsedLocation = {
  state: { route: { destination1: 'Remera', destination2: 'Kacyiru' } }
};
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockImplementation(() => {
    return mockedUsedLocation;
  })
}));

describe('View Page', () => {
  it('should render the View component', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <RouteView />
        </MemoryRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render the Map MapContainer component', () => {
    const wrapper = mount(
      <MemoryRouter>
        <RouteView />
      </MemoryRouter>
    );
    expect(wrapper.find('MapContainer').length).toEqual(1);
  });
});
