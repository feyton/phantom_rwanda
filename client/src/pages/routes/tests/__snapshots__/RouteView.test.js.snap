// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Page should render the View component 1`] = `
<div
  className="p-4 bg-white w-[90vh] font-raleway"
>
  <h2
    className="text-2xl text-center font-bold"
  >
    Route informations
  </h2>
  <div
    className="mt-5"
  >
    <ul>
      <li>
        <b>
           Origin: 
        </b>
         
      </li>
      <li>
        <b>
           Destination: 
        </b>
         
      </li>
    </ul>
  </div>
  <div
    className=" h-[300px] overflow-hidden py-2"
  >
    <h1
      className="font-raleway font-bold"
    >
       Map 
    </h1>
     
    <hr />
    <div
      className="h-full w-full mb-2"
    />
  </div>
</div>
`;
